services/models.py

```diff
@@ -183,22 +258,53 @@ def delete_gdpr_data(self, api_token: str, dry_run=False):
         profile on the related GDPR API.
 
         Dry run parameter can be used for asking the service if delete is possible.
-        An exception will be raised by this method if deletion response from the
-        service indicates an error or if GDPR related URLs have not been configured
-        for the related service.
+        Returns a dict with keys "service", "dry_run", "success", and "errors".
+        The errors content from the service is returned if the service provides a JSON
+        response with an "errors" key containing valid error content.
```

Comment:
Could this use a more strict type, perhaps a dataclass or something like that? That would allow IDEs to help fill in the correct attributes etc. So instead of writing the structure into a comment, write it as code?